packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -215,6 +215,13 @@ export abstract class Backend {
     return new VirtualServiceBackend(virtualService, props.tlsClientPolicy);
   }
 
+  /**
+   * Construct a Virtual Service backend via a Virtual Service name
+   */
+  public static virtualServiceName(virtualServiceName: string, props: VirtualServiceBackendOptions = {}): Backend {
+    return new VirtualServiceNameBackend(virtualServiceName, props.tlsClientPolicy);
+  }
```

Comment:
I don't love this API. In my opinion, it's kind of confusing, easy to misuse (for example, what if I do `Backend.virtualServiceName(virtualService.virtualServiceName)`? This will still not work), and the docs do not explain at all what problem does this method solve.

I'd prefer if we added a new property to `IVirtualService`, let's call it `virtualServicePhysicalName`, and fill it with the simple string name (_not_ with a Token value that results in `{ Ref: 'VirtualServiceLogicalId' }` being rendered in the template). Then, change the API here to take an `IVirtualService` instead of a `string`.

Let's also make sure to explain in the JSDocs of this method how is it different than `Backend.virtualService()`.

Reply:
I believe using `Backend.virtualServiceName(virtualService.virtualServiceName)` would offer the same functionality that currently exists, but would not add the dependency automatically. The user would be required to specify that the Virtual Node depends on that Virtual Service.

Won't the token value still be needed if the new `physicalName` field is added? The `virtualServiceName` is currently an optional field in `IVirtualService`, but required by the App Mesh service. Therefore it, and the `physcialName` field, will be have a fallback to be generated using `cdk.Lazy.string({ produce: () => cdk.Names.uniqueId(this) })`.

Reply:
> I believe using `Backend.virtualServiceName(virtualService.virtualServiceName)` would offer the same functionality that currently exists, but would not add the dependency automatically. The user would be required to specify that the Virtual Node depends on that Virtual Service.

No, it would not work (which makes sense, because `virtualService.virtualServiceName` is exactly how the current `Backend.virtualService()` method is implemented - so, that would mean this new `Backend.virtualServiceName()` method is not needed!). That's because `virtualService.virtualServiceName` results in a `{ 'Fn::Get': 'VirtualServiceLogicalId.VirtualServiceName' }` being rendered in the resulting CloudFormation template, and that creates a dependency order (the resource with  `{ 'Fn::Get': 'VirtualServiceLogicalId.VirtualServiceName' }` will have a dependency on `VirtualServiceLogicalId`). So, the cycle will still happen.

> Won't the token value still be needed if the new `physicalName` field is added? The `virtualServiceName` is currently an optional field in `IVirtualService`, but required by the App Mesh service. Therefore it, and the `physcialName` field, will be have a fallback to be generated using `cdk.Lazy.string({ produce: () => cdk.Names.uniqueId(this) })`.

I was a little imprecise with my wording. What I meant is that `virtualServicePhysicalName` is also a Token, but one that doesn't resolve to a CloudFormation expression like `virtualServiceName` does, but a Token that resolves to a specific string - the name of the VirtualService (either assigned by the customer using the `virtualServiceName` property when creating the `VirtualService`, or generated by the CDK).